### ISSUE TITLE
add class level annotation to a test with javaassist

### ISSUE THREAD

**Issue by chaudharydeepak** [jboss-javassist/javassist#380]
Hi,

I am trying to add class level annotation to a test class as following -
` @org.testng.annotations.Listeners(value = org.deployd.agent.TestNgListener.class)`
   `public class FirstSeleniumTest {`

In `transform` method, logic is as follows

                ClassPool classPool = ClassPool.getDefault();
                classPool.appendClassPath(new LoaderClassPath(loader));
                CtClass ctClass;
                try {
                ctClass = classPool.get(className.replaceAll("/", "."));
                if (ctClass.isFrozen()) {
                    ctClass.defrost();
                }
                ClassFile classFile = ctClass.getClassFile();
                ConstPool constPool = classFile.getConstPool();
                AnnotationsAttribute annotationsAttribute = new AnnotationsAttribute(constPool, AnnotationsAttribute.visibleTag);
                Annotation annotation = new Annotation(Listeners.class.getName(), constPool);
                annotation.addMemberValue("value", new ClassMemberValue("org/deployd/agent/TestNgListener", constPool));
                annotationsAttribute.addAnnotation(annotation);
                ctClass.getClassFile().addAttribute(annotationsAttribute);
                ClassFilePrinter.print(ctClass.getClassFile());
                byteCode = ctClass.toBytecode();
                ctClass.detach();
`

when I run my `FirstSeleniumTest` with above agent implementation, the exception as below is thrown -

	java.lang.annotation.AnnotationTypeMismatchException: Incorrectly typed data found for annotation element 
        public abstract java.lang.Class[] org.testng.annotations.Listeners.value() (Found data of type class java.lang.Class[class 	org.deployd.agent.TestNgListener])
	at sun.reflect.annotation.AnnotationTypeMismatchExceptionProxy.generateException(AnnotationTypeMismatchExceptionProxy.java:57)
	at sun.reflect.annotation.AnnotationInvocationHandler.invoke(AnnotationInvocationHandler.java:84)
	at com.sun.proxy.$Proxy5.value(Unknown Source)
	at org.testng.internal.annotations.JDK15TagFactory.createListenersTag(JDK15TagFactory.java:462)
	at org.testng.internal.annotations.JDK15TagFactory.createTag(JDK15TagFactory.java:64)
	at org.testng.internal.annotations.JDK15AnnotationFinder.lambda$findAnnotation$0(JDK15AnnotationFinder.java:279)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1688)
	at org.testng.internal.annotations.JDK15AnnotationFinder.findAnnotation(JDK15AnnotationFinder.java:278)
	at org.testng.internal.annotations.JDK15AnnotationFinder.findAnnotation(JDK15AnnotationFinder.java:198)
	at org.testng.internal.TestListenerHelper.findAllListeners(TestListenerHelper.java:119)
	at org.testng.TestRunner.initListeners(TestRunner.java:360)
	at org.testng.TestRunner.init(TestRunner.java:342)
	at org.testng.TestRunner.init(TestRunner.java:292)
	at org.testng.TestRunner.<init>(TestRunner.java:183)
	at org.testng.SuiteRunner$DefaultTestRunnerFactory.newTestRunner(SuiteRunner.java:596)
	at org.testng.SuiteRunner.init(SuiteRunner.java:173)
	at org.testng.SuiteRunner.<init>(SuiteRunner.java:107)
	at org.testng.TestNG.createSuiteRunner(TestNG.java:1300)
	at org.testng.TestNG.createSuiteRunners(TestNG.java:1276)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1125)
	at org.testng.TestNG.runSuites(TestNG.java:1063)
	at org.testng.TestNG.run(TestNG.java:1031)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)

any pointers to what I might be doing wrong here? Thank you for your help.

**Comment by chaudharydeepak** [comment]
on further analysis, I found the annotation value excepts an array vs single value of the type. I was able to resolve this via javassist as following:

    AnnotationsAttribute annotationsAttribute = new AnnotationsAttribute(constPool, AnnotationsAttribute.visibleTag);
    Annotation annotation = new Annotation(Listeners.class.getName(), constPool);
    MemberValue[] memberValues = new MemberValue[]{new ClassMemberValue(TestNgListener.class.getName(), constPool)};
    ArrayMemberValue arrayMemberValue = new ArrayMemberValue(constPool);
    arrayMemberValue.setValue(memberValues);
    annotation.addMemberValue("value", arrayMemberValue);
    annotationsAttribute.addAnnotation(annotation);
    ctClass.getClassFile().addAttribute(annotationsAttribute);
